src/cm_utils.js

```diff
@@ -0,0 +1,64 @@
+import { addClass, elt, removeChildren, rmClass } from "./util/dom";
+import { on } from "./util/event";
+import { clearCaches } from "./measurement/position_measurement";
+import { setScrollLeft, setScrollTop } from "./scroll_events";
+import { scrollbarModel } from "./display/scrollbars";
+import { updateGutterSpace } from "./display/update_display";
+import { indexOf } from "./util/misc";
+
+export function initScrollbars(cm) {
+  if (cm.display.scrollbars) {
+    cm.display.scrollbars.clear();
+    if (cm.display.scrollbars.addClass)
+      rmClass(cm.display.wrapper, cm.display.scrollbars.addClass);
+  }
+
+  cm.display.scrollbars = new scrollbarModel[cm.options.scrollbarStyle](function(node) {
+    cm.display.wrapper.insertBefore(node, cm.display.scrollbarFiller);
+    // Prevent clicks in the scrollbars from killing focus
+    on(node, "mousedown", function() {
+      if (cm.state.focused) setTimeout(function() { cm.display.input.focus(); }, 0);
+    });
+    node.setAttribute("cm-not-content", "true");
+  }, function(pos, axis) {
+    if (axis == "horizontal") setScrollLeft(cm, pos);
+    else setScrollTop(cm, pos);
+  }, cm);
+  if (cm.display.scrollbars.addClass)
+    addClass(cm.display.wrapper, cm.display.scrollbars.addClass);
+}
+
+export function themeChanged(cm) {
+  cm.display.wrapper.className = cm.display.wrapper.className.replace(/\s*cm-s-\S+/g, "") +
+    cm.options.theme.replace(/(^|\s)\s*/g, " cm-s-");
+  clearCaches(cm);
+}
+
+// Rebuild the gutter elements, ensure the margin to the left of the
+// code matches their width.
+export function updateGutters(cm) {
```

Comment:
Maybe a display/gutters.js for this one and the one below? That leaves only `themeChanged` in this file, which you could smuggle into some other relatively generic file to drop this module entirely.